views/forms/intake_packet.php

```diff
@@ -247,120 +440,324 @@
     $datestamp = date("Y-m-d");
     $eID = $_SESSION['employeeid'];
 
-    // Stored Procedures
+    if (isset($params[0]) && isset($params[1]) && isset($params[2])) {
+
+        /*************
+         *  UPDATES  *
+         *************/
+
+        $updatePeopleResult = $db->query("UPDATE
+                                    People
+                                    SET
+                                    firstName = $1,
+                                    lastName = $2,
+                                    middleInit = $3
+                                    WHERE 
+                                    peopleID = $4;", [$intake_firstname, $intake_lastname, $intake_middlein, $params[1]]);
+
+        $updateParticipantResult = $db->query("UPDATE 
+                                    Participants
+                                    SET 
+                                    dateOfBirth = $1,
+                                    race = $2,
+                                    sex = $3
+                                    WHERE 
+                                    participantID = $4;", [$intake_dob, $intake_ethnicity, $intake_sex, $params[1]]);
+
+        $newZip = $db->query("INSERT INTO ZipCodes VALUES($1, $2, $3);", [$intake_zip, $intake_city, $intake_state]);
+
+        $updateAddressResult = $db->query("UPDATE
+                                    Addresses
+                                    SET
+                                    addressNumber = $1,
+                                    aptInfo = $2,
+                                    street = $3,
+                                    zipCode = $4
+                                    WHERE
+                                    addressID = $5;", [$intake_street_num, $intake_intake_apt_info, $intake_street_name, $intake_zip, $params[2]]);
+
+
+        $updateDayPhoneResult = $db->query("UPDATE 
+                                    FormPhoneNumbers
+                                    SET 
+                                    phoneNumber = $1
+                                    WHERE
+                                    formID = $2 AND
+                                    phoneType = $3;", [$intake_phone_day, $params[2], 'Day']);
```

Comment:
If I never set my phone number in the first place and  then I went back to edit the form and added my phone number, there would be no pre-existing 'Day' phone in the database to update. This is the case with the other types of phones. You should do an "upsert" (update-insert) where you update if it exists, otherwise insert a new row.

Reply:
Fixed on intake and self referral, thanks for the catch!